internal/serve/middleware/middleware.go

```diff
@@ -176,6 +178,69 @@ func (c cspItem) String() string {
 	return fmt.Sprintf("%s %s;", c.ContentType, strings.Join(c.Policy, " "))
 }
 
+// LoggingMiddleware is a middleware that logs requests to the logger.
+func LoggingMiddleware() func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+			mw := mutil.WrapWriter(rw)
+
+			reqCtx := req.Context()
+			tenant, err := tenant.GetTenantFromContext(reqCtx)
+			if err != nil {
+				httperror.InternalError(reqCtx, "Cannot find tenant from context", err, nil).Render(rw)
+				return
+			}
```

Comment:
Are you sure you want to make the logging middleware fail if the tenant is not in the context?